documentation/homework/week4/src/stores/todo_actions.js

```diff
@@ -0,0 +1,150 @@
+import { observable, action, runInAction } from 'mobx'
+
+// Setting our localStorage
+// import { locals } from '../utils'
+
+// set the default state of the items Object
+const default_item_state = {
+  description: '',
+  deadline: '',
+}
+
+async function request(url, method, inputToString) {
+  const params = {
+    "body": JSON.stringify(inputToString),
+    "headers": {
+      "Content-type": "application/json"
+    }
+  }
+
+  let options = {}
+  method = method ? method.toUpperCase() : "GET"
+  switch (method) {
+    case "POST":
+    case "PATCH":
+      options = { method, ...params, }
+      break
+    default:
+      options = { method }
+  }
+
+  // is't Okay to catch the errors here with in 
```

Comment:
Not entirely sure what you are asking but I'll respond in general:

Errors "bubble". If you don't catch them inside your `request` helper, you'll need to catch them one level up, eg wherever you call `request()`. Either way is fine, but typically you'd expect a helper like `request` to pass errors up through invocation, not handle them itself. Imagine we were using `request` in multiple stores, and each store wanted to handle errors in a slightly different way.

The only hard rule is that you _must_ catch errors somewhere. Uncaught rejections will soon be a breaking runtime error in Node.

Reply:
yeah thank you Isaac I got it 🙂